crypto++ does not build on OS X

[Full log here](https://gist.github.com/tel/5b71e1557f27093f1463).

The short of it is that `libtool` doesn't have a `-static` flag on OS X.

```
$ nix-env -iA nixpkgs.cryptopp
installing ‘crypto++-5.6.2’
these derivations will be built:
  /nix/store/8ygzvyzchxx9bnc7a0xhpxw8rgpfp027-crypto++-5.6.2.drv

[...]

libtool -static -o libcryptopp.a shacal2.o seed.o shark.o zinflate.o gf2n.o socketft.o oaep.o cryptlib_bds.o default.o wait.o wake.o twofish.o safer.o iterhash.o adler32.o marss.o blowfish.o ecp.o strciphr.o dh2.o ida.o zlib.o elgamal.o crc.o algparam.o gf2_32.o tea.o rijndael.o eax.o network.o sha.o emsa2.o pkcspad.o squaretb.o gcm.o idea.o authenc.o cmac.o xtrcrypt.o simple.o mars.o rc5.o fipstest.o queue.o hrtimer.o vmac.o eprecomp.o winpipes.o polynomi.o dsa.o fips140.o gzip.o dessp.o files.o base32.o sharkbox.o randpool.o esign.o hex.o hmac.o osrng.o skipjack.o integer.o xtr.o sosemanuk.o cpu.o filters.o bfinit.o rabin.o 3way.o rdtables.o rsa.o cbcmac.o pch.o tftables.o gost.o md5.o nbtheory.o panama.o sha3.o modes.o casts.o algebra.o cryptlib.o gfpcrypt.o dll.o ec2n.o blumshub.o des.o basecode.o zdeflate.o base64.o rc6.o gf256.o mqueue.o misc.o pssr.o channels.o rng.o tiger.o cast.o rw.o square.o asn.o whrlpool.o md4.o dh.o ccm.o md2.o mqv.o tigertab.o rc2.o ttmac.o luc.o seal.o salsa.o trdlocal.o pubkey.o camellia.o ripemd.o arc4.o eccrypto.o serpent.o
building adhoc.o
libtool: unrecognized option `-static'
libtool: Try `libtool --help' for more information.
make: *** [libcryptopp.a] Error 1
make: *** Waiting for unfinished jobs....
builder for ‘/nix/store/8ygzvyzchxx9bnc7a0xhpxw8rgpfp027-crypto++-5.6.2.drv’ failed with exit code 2
error: build of ‘/nix/store/8ygzvyzchxx9bnc7a0xhpxw8rgpfp027-crypto++-5.6.2.drv’ failed
```

--- a/pkgs/development/libraries/crypto++/GNUmakefile.patch
+++ b/pkgs/development/libraries/crypto++/GNUmakefile.patch
@@ -1,0 +1,13 @@
+--- crypto++/GNUmakefile 2013-02-20 10:30:52.000000000 -0500
++++ crypto++/GNUmakefile 2015-05-07 18:34:25.000000000 -0500
+@@ -87,8 +87,8 @@
+ endif
+
+ ifeq ($(UNAME),Darwin)
+-AR = libtool
+-ARFLAGS = -static -o
++AR = ar
++ARFLAGS = cru
+ CXX = c++
+ IS_GCC2 = $(shell $(CXX) -v 2>&1 | $(EGREP) -c gcc-932)
+ ifeq ($(IS_GCC2),1)

--- a/pkgs/development/libraries/crypto++/default.nix
+++ b/pkgs/development/libraries/crypto++/default.nix
@@ -8,7 +8,8 @@ stdenv.mkDerivation rec {
     sha256 = "0x1mqpz1v071cfrw4grbw7z734cxnpry1qh2b6rsmcx6nkyd5gsw";
   };
 
-  patches = stdenv.lib.optional (stdenv.system != "i686-cygwin") ./dll.patch;
+  patches = (stdenv.lib.optional (stdenv.system != "i686-cygwin") ./dll.patch)
+            ++ (stdenv.lib.optional stdenv.isDarwin ./GNUmakefile.patch);
 
   buildInputs = [ unzip libtool ];
 

